.github/workflows/api.yaml

```diff
@@ -36,23 +41,34 @@ jobs:
         uses: actions/checkout@master
         with:
           fetch-depth: 0
+          # Check https://github.com/livepeer/go-livepeer/pull/1891
+          # for ref value discussion
+          ref: ${{ github.event.pull_request.head.sha }}
 
-      - uses: actions/setup-node@v2
+      - uses: actions/setup-node@v3
         with:
           node-version: "16"
           cache: "yarn"
 
       - name: Tags
         id: tags
-        uses: livepeer/action-gh-release-tags@v0
+        uses: livepeer/action-gh-release-tags@latest
 
-      - name: yarn install
-        run: |
-          yarn install --frozen-lockfile
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
 
-      - name: yarn test
-        run: |
-          yarn test
+      - uses: actions/cache@v3
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
```

Comment:
Doesn't `actions/setup-node` already handle deps caching with the `cache: yarn` arg?

Yeah I checked on the builds in this branch, and now there are 2 steps doing possibly redundant caching:
<img width="1370" alt="Screen Shot 2022-04-11 at 19 03 58" src="https://user-images.githubusercontent.com/1613383/162840995-380c676e-84d0-4b4f-884a-2a900712bad1.png">